### DISCUSSION TITLE
Render Configuration Context for more objects than just Devices and Virtual Machines

### DISCUSSION THREAD

**Discussion by lbreitk** [netbox-community/netbox#16294] [Ideas]
# Summary
Could we add the ability to render configuration context for objects other than just devices and virtual machines? For 
example, unless we add **tons** of custom fields, [this](#example-config) sort of thing doesn't make sense to replace with custom fields, but we'd like to be able to utilize it as a Configuration Management Database. E.G. I am using Netbox to generate a Kea DHCP configuration, however it's not something you'd want to do per-device, since some configurations are more global - like the next server for DHCP for example.

# Example Config
```json
{
    "dhcp": {
        "file_server": {
            "server": "192.0.2.1",
            "tls": false
        },
        "next_server": "192.0.2.1"
    },
    "dns": {
        "domain": "domain.example.com"
    },
    "info": {
        "documentation": "https://wiki.example.com/tenant",
        "email": "group@example.com"
    },
    "secrets": {
        "service_accounts": [
            {
                "name": "account1",
                "secret_type": "secret server",
                "access_link": "https://example.secretservercloud.com/app/#/secrets/1000/general"
            },
            {
                "name": "account2",
                "secret_type": "vault",
                "access_link": "https://vault..example.com/ui/vault/secrets/kv/show/tenant/service_accounts/account2"
            }
        ]
    },
    "jumpbox": {
        "device": "server1",
        "public_ip": "198.51.100.1"
    },
    "ldap": {
        "allow_groups": [],
        "domain": "",
        "home_server": "",
        "type": "none"
    },
    "meta": {
        "last_updated": "2024-05-24",
        "schema_version": "0.2.0",
        "source": "manual"
    },
    "services": {
        "consul": {
            "ips": [],
            "pools": []
        },
        "dns": {
            "ips": [
                "192.0.2.1",
                "192.0.2.2"
            ],
            "pools": []
        },
        "http_file": {
            "ips": [
                "192.0.2.1"
            ],
            "pools": []
        },
        "nfs": {
            "ips": [],
            "pools": []
        },
        "ntp": {
            "ips": [
                "192.0.2.1",
                "192.0.2.2"
            ],
            "pools": [
                "ntp.example.com"
            ]
        },
        "snmp": {
            "ips": [],
            "pools": []
        },
        "syslog": {
            "ips": [],
            "pools": []
        },
        "tftp": {
            "ips": [
                "192.0.2.1"
            ],
            "pools": []
        }
    },
    "time": {
        "skew": {
            "max": 10
        },
        "timezone": "Etc/UTC"
    }
}
```

In this case, it makes more sense to render the configuration context at the tenant level (we are using the tenant field to seperate different environments), rather than try to iterate over all devices in tenant and then generate different global configs per-device. As of the moment, I'll have to just query the configuration context and filter for the tenant, and probably later write some custom code to handle weights. 

# Other thoughts
* Last I checked, services don't have the ability to assign them to tenants. If they did, I could replace the services object with iterating over all services defined for a particular tenant.
* Some of this stuff doesn't make sense to replace with a custom field - there may be specific servers that have a reason to use a different timezone, or maybe an application. But there's not reason to  handle that at the DHCP level. DHCP could pull the tenant-level timezone, with ansible pulling from the device-level configuration context later to set the per-device timezone, if needed

**Comment by empusas** [comment]
I would like to see config context for virtual context and virtual chassis.
